apps/backend/src/modules/project-groups/project-group.repository.ts

```diff
@@ -0,0 +1,70 @@
+import { transaction } from "objection";
+
+import { changeCase } from "~/libs/helpers/helpers.js";
+import { type Repository } from "~/libs/types/types.js";
+
+import { ProjectGroupEntity } from "./project-group.entity.js";
+import { type ProjectGroupModel } from "./project-group.model.js";
+
+class ProjectGroupRepository implements Repository {
+	private projectGroupModel: typeof ProjectGroupModel;
+
+	public constructor(projectGroupModel: typeof ProjectGroupModel) {
+		this.projectGroupModel = projectGroupModel;
+	}
+
+	public async create(entity: ProjectGroupEntity): Promise<ProjectGroupEntity> {
+		const { name, permissionIds, projectId, userIds } = entity.toNewObject();
+		const key = changeCase(name, "snakeCase");
+
+		const trx = await transaction.start(this.projectGroupModel.knex());
+
+		const projectGroupData = {
+			key,
+			name,
+			permissions: permissionIds,
+			projects: projectId,
+			users: userIds,
+		};
+
+		const createdProjectGroup = await this.projectGroupModel
+			.query(trx)
+			.insertGraph(projectGroupData, { relate: true })
+			.returning("*")
+			.withGraphJoined("[permissions, projects, users]");
+
+		await trx.commit();
```

Comment:
Hm, I think this syntax requires us to wrap everything into a try/catch and rollback the migration in case of error, doesn't it?